lightning-persister/src/lib.rs

```diff
@@ -28,9 +27,38 @@ use lightning::ln::channelmanager::ChannelManager;
 use lightning::util::logger::Logger;
 use lightning::util::ser::{ReadableArgs, Writeable};
 use std::fs;
-use std::io::{Cursor, Error};
+use std::io::{Cursor, Error, Write};
 use std::ops::Deref;
-use std::path::{Path, PathBuf};
+use std::path::{Path, PathBuf, MAIN_SEPARATOR};
+
+
+/// Trait that handles persisting a [`ChannelManager`] and [`NetworkGraph`] to disk.
+pub trait Persister
+{
+	/// Persist the given [`ChannelManager`] to disk, returning an error if persistence failed
+	/// (which will cause the BackgroundProcessor which called this method to exit).
+	fn persist_manager<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>(&self, channel_manager: &ChannelManager<Signer, M, T, K, F, L>) -> Result<(), std::io::Error> where
+	M::Target: 'static + chain::Watch<Signer>,
```

Comment:
Hmm, what is the error if you keep the types on the `Persister` trait? It does seem like a single instance of a `Persister` should really only care about persisting one type of `ChannelManager`, not multiple.

Reply:
The error is on the test struct Persister that is used in the BackgroundProcessor tests:

```
error[E0284]: type annotations needed
   --> lightning-background-processor/src/lib.rs:440:39
    |
440 |                 None => self.filesystem_persister.persist_graph(network_graph),
    |                                                   ^^^^^^^^^^^^^ cannot infer type for type parameter `M` declared on the trait `Persister`
    |
    = note: cannot satisfy `<_ as Deref>::Target == _`
```

Reply:
Being explicit solves the issue for me - 
`LPPersister::<Signer, M, T, K, F, L>::persist_graph(&self.filesystem_persister, network_graph)`

Reply:
ended up just removing this whole section of code since it's unnecessary to explicitly impl Persister anymore.  Just implemented KVStorePersister here and handled throwing errors based on the key